### PR TITLE
Added option to display icons instead of text in context actions for iOS

### PR DESCRIPTION
**Pull request by Sebastian1989101**
### Description of Change ###

/!\ This is the same PR as https://github.com/xamarin/Xamarin.Forms/pull/6363 just implemented a different (maybe much cleaner?) way /!\
 
I've added a enum to choose if ContextActions within a ListView should be displayed as Icon or Text. 

### Issues Resolved ### 

None

### API Changes ###

Added:
 - Added On().SetContextActionDisplay(ContextActionDisplay.Icon) for Cell

### Platforms Affected ### 

- iOS

### Behavioral/Visual Changes ###

None with default settings. When ContextActionDisplay.Icon is set for parent Cell, all ContextActions for this Cell will appear as Icon instead of text in iOS.

### Before/After Screenshots ### 

Not applicable

### Testing Procedure ###

Added "Context Action Display Icon/Text" to Control Gallery which includes ContextActions Image or Text based.

### PR Checklist ###

- [ ] Has automated tests 
- [x] Rebased on top of the target branch at time of PR
- [ ] Changes adhere to coding standard

### Additional noted ###

This currently uses a hacky way to obtain the system more icon which may creates issues - I was not sure where I should add such icons if I recreate them myself. 

/!\ This is the same PR as https://github.com/xamarin/Xamarin.Forms/pull/6363 just implemented a different (maybe much cleaner?) way /!\

### CODE REVIEW

File: Xamarin.Forms.Platform.iOS/ContextActionCell.cs [hunk 1]
```diff
@@ -641,6 +679,20 @@ UIView SetupButtons(nfloat width, nfloat height)
 			return null;
 		}
 
+		//Should this really be this cheaty? Or is there a better way to obtain the native more icon?
+		//Note: Using icons improperly may violate iOS Human Interface Guidelines and cause the app to be rejected from App Store
+		UIImage GetMoreImage()
+		{
+			var tabBar = new UITabBar();
```
Comment (hartez) [inline Xamarin.Forms.Platform.iOS/ContextActionCell.cs]: You can do this without creating a new UITabBar:

```
var item = new UITabBarItem(UITabBarSystemItem.More, 0);
var img = (item != null) ? UIImage.FromImage(item.SelectedImage.CGImage, item.SelectedImage.CurrentScale, item.SelectedImage.Orientation) : new UIImage();
return img.ImageWithRenderingMode(UIImageRenderingMode.AlwaysTemplate);
```

File: Xamarin.Forms.Platform.iOS/ContextActionCell.cs [hunk 2]
```diff
@@ -641,6 +679,20 @@ UIView SetupButtons(nfloat width, nfloat height)
 			return null;
 		}
 
+		//Should this really be this cheaty? Or is there a better way to obtain the native more icon?
```
Comment (hartez) [inline Xamarin.Forms.Platform.iOS/ContextActionCell.cs]: I don't think there's a better way to get the icon until iOS 13.

File: Xamarin.Forms.Platform.iOS/ContextActionCell.cs [hunk 3]
```diff
@@ -641,6 +679,20 @@ UIView SetupButtons(nfloat width, nfloat height)
...
+		UIImage GetMoreImage()
+		{
+			var tabBar = new UITabBar();
+			var item = new UITabBarItem(UITabBarSystemItem.More, 0);
+			tabBar.Items = new[] { item };
+
+			var itemView = tabBar.Subviews[0];
+			var imageView = itemView.Subviews[0] as UIImageView;
+
+			return imageView?.Image.ImageWithRenderingMode(UIImageRenderingMode.AlwaysTemplate);
```
Comment (hartez) [inline Xamarin.Forms.Platform.iOS/ContextActionCell.cs]: Also, once you've retrieved the image you should cache it so every cell doesn't have to do this.

File: Xamarin.Forms.Core/PlatformConfiguration/iOSSpecific/Cell.cs [hunk 4]
```diff
@@ -6,6 +6,8 @@ public static class Cell
 	{
 		public static readonly BindableProperty DefaultBackgroundColorProperty = BindableProperty.Create(nameof(DefaultBackgroundColor), typeof(Color), typeof(Cell), Color.Default);
 
+		public static readonly BindableProperty ContextActionDisplayProperty = BindableProperty.Create(nameof(ContextActionDisplay), typeof(ContextActionDisplay), typeof(Cell), ContextActionDisplay.Text);
```
Comment (hartez) [inline Xamarin.Forms.Core/PlatformConfiguration/iOSSpecific/Cell.cs]: `typeof(Cell)` will refer to this static class; this should use `typeof(FormsElement)`.

**Comment by jsuarezruiz** [comment]
@Sebastian1989101 This looks very nice bu there is a conflict and some comments, could you check it?. Thanks!.

**Comment by Sebastian1989101** [comment]
@jsuarezruiz I've planed to fix my PR's as soon as I have time again. Currently my spare time is pretty limited due to marriage and first child beside the normal job. If someone wants this sooner, it can easily be adopted into a own PR or adjusted right here.

**Comment by PureWeen** [comment]
Closing this PR for now in favor of this one

https://github.com/xamarin/Xamarin.Forms/pull/6363#pullrequestreview-264450960

**Pull request closed without merging**